engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/AnalyzeModuleInScopeJob.scala

```diff
@@ -0,0 +1,140 @@
+package org.enso.interpreter.instrument.job
+
+import org.enso.compiler.context.{
+  ExportsBuilder,
+  ModuleExportsDiff,
+  SuggestionBuilder,
+  SuggestionDiff
+}
+import org.enso.interpreter.instrument.execution.RuntimeContext
+import org.enso.interpreter.runtime.Module
+import org.enso.pkg.QualifiedName
+import org.enso.polyglot.{ModuleExports, Suggestion}
+import org.enso.polyglot.data.Tree
+import org.enso.polyglot.runtime.Runtime.Api
+
+import java.util.logging.Level
+
+final class AnalyzeModuleInScopeJob(
+  moduleName: Option[QualifiedName],
+  modules: Iterable[Module]
+) extends Job[Unit](
+      List(AnalyzeModuleJob.backgroundContextId),
+      false,
+      false
+    ) {
+
+  private val exportsBuilder = new ExportsBuilder
+
+  /** @inheritdoc */
+  override def run(implicit ctx: RuntimeContext): Unit = {
+    // There are two runtime flags that can disable suggestions for project
+    // and global modules (libraries). They are used primarily in tests to
+    // disable the suggestion updates and reduce the number of messages that
+    // runtime sends.
+    if (ctx.executionService.getContext.isProjectSuggestionsEnabled) {
+      // When the project module is compiled it can involve compilation of
+      // global (library) modules, so we need to check if the global
+      // suggestions are enabled as well.
+      if (ctx.executionService.getContext.isGlobalSuggestionsEnabled) {
+        modules.foreach(analyzeModuleInScope)
+        ctx.endpoint.sendToClient(
+          Api.Response(Api.AnalyzeModuleInScopeJobFinished())
+        )
+      } else {
+        // When the global suggestions are disabled, we will skip indexing
+        // of external libraries, but still want to index the modules that
+        // belongs to the project.
```

Comment:
I am not sure I understand the terminology well enough, however this comment sounds like user story [#182261861](https://www.pivotaltracker.com/n/projects/2539304/stories/182261861) request! Ideally:
- first of all provide the IDE information about _modules that belong to the project_
- only then (and in background) deliver the suggestions for the _global modules_

Shall we have two versions of this class? One delivering this "else branch" asap. Another delivering global suggestions "later"?